### ISSUE TITLE
Auto-populate router entries and nav list

### ISSUE THREAD

**Issue by jtkese** [ember-learn/ember-cli-addon-docs#199]
Hey,

I've been using ember-cli-addon-docs for a project, it has been great for developing isolated components. Have found that there's a fair amount of boilerplate when adding something to the docs and thinking much of it can be automated.

Filtering the keys from requirejs.entries can yield all components/mixins/etc with addon doc md files, which can then be used to populate docsRoute entries in the router and build the navigation handlebars template for the docs viewer.

A method like autopopulateDocRoutes() would be nice to be able to add to a docsRoute() call to populate it with routes based on such auto-detection. For the template, a sub-component to viewer.nav like "nav.autopopulate" would also be fantastic to insert nav.section elements (for directories or types) and nav.item elements--inserted inline, rather than something replacing viewer.nav entirely, so it can appear alongside any other custom sections and items not represented by md files.

I built this functionality into a project to reduce trivial work and the number of merge conflicts when multiple team members are adding components to the docs, would love to see such a thing included by default.

Cheers

**Comment by samselikoff** [comment]
I agree something like this should make its way into AD. I wanted to start with "just components" so folks' docs apps are "just Ember apps", but given most (all?) people will be using the DocsViewer component I think some higher-level abstractions would be nice. 👍
